src/devices/src/virtio/queue.rs

```diff
@@ -104,9 +104,9 @@ impl<'a> DescriptorChain<'a> {
         // These reads can't fail unless Guest memory is hopelessly broken.
         let desc = match mem.read_obj::<Descriptor>(desc_head) {
             Ok(ret) => ret,
-            Err(_) => {
+            Err(e) => {
                 // TODO log address
-                error!("Failed to read from memory");
+                error!("Failed to read virtio descriptor from memory: {}", e);
                 return None;
             }
```

Comment:
Prefer `err` over `e`.
```suggestion
            Err(err) => {
                // TODO log address
                error!("Failed to read from memory");
                error!("Failed to read virtio descriptor from memory: {}", err);
                return None;
            }
```